spec/controllers/admin/videos_controller_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+
+describe Admin::VideosController do
+  it {is_expected.to use_before_action(:require_user)}
+  
+  describe "GET new" do
+    before {set_current_user}
```

Comment:
This before is the start of a poorly formed block. You set the before (which should be necessary for every test in the block) and then override it with `set_current_admin` and then don't really use it in either of the final tests. There are a couple of ways to restructure this. Either drop the before block and simply write out the tests in full or use a context for each type of user. In this case (especially with your `it_behaves_like`) I would probably simply shift `set_current_user` to line 15. If it were something larger I would consider a context.

Regardless the most important thing is that every line matters. Using a `before` applies to every test in the block and should therefore be required for every test--if it doesn't don't use it or restructure the offending test out. Use techniques such as `before` to trim away fat, but don't allow their abuse to muddy the waters.